### PR TITLE
docs(guide): Add Custom Backend guide

### PR DESCRIPTION
**Pull request by francoischalifour**
This is the same content as algolia/instantsearch.js#2943 with React code samples in "Using the search client with InstantSearch".

[:link: **Link to the guide**](https://deploy-preview-1290--react-instantsearch.netlify.com/react-instantsearch/guide/custom_backend)

### CODE REVIEW

**Comment by samouss** [comment]
Since a couple of days / weeks Argos keeps failing with the same screenshots. In this PR we don't update the UI but the reports still fail. Any ideas? cc @oliviertassinari

**Comment by oliviertassinari** [comment]
@samouss Are you referring to the diffs like this one?
capture d ecran 2018-06-01 a 13 28 32

File: docgen/src/guide/Custom_backend.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+You’ll be able to search within your refinement lists from your custom backend.
+
+> Note that you will need to set the option [`searchForFacetValues`](https://community.algolia.com/instantsearch.js/v2/widgets/refinementList.html#struct-RefinementListWidgetOptions-searchForFacetValues) to `true` in your refinement list to make it searchable.
+
+## Using the search client with InstantSearch
+
+Now, we need to tell InstantSearch to use the search client that we’ve created. This is possible with the `searchClient` option. This parameter will disable all Algolia requests coming from the frontend and will proxy them to your own backend implementation.
+
+```javascript
+const App = () =>
```
Comment (samouss) [inline docgen/src/guide/Custom_backend.md]: Add some parens around the body of the function.

**Comment by samouss** [comment]
Yes indeed, before we didn't have those diffs.

File: docgen/src/guide/Custom_backend.md (hunk 1 above)
Comment (francoischalifour) [reply docgen/src/guide/Custom_backend.md]: That's what I would have done too. Although I copy-pasted the code snippet from the documentation (which I can't find now).

Anyway, I'm for parens, will update.

**Comment by samouss** [comment]
Ok thanks a lot for your inputs, I'll investigate more on that subject.

**Comment by Haroenv** [comment]
I think we could just hide the images altogether, replace them with `x` or something

**Comment by oliviertassinari** [comment]
@Haroenv I agree!

**Comment by samouss** [comment]
Indeed, thanks for the suggestions!

**Pull request merged**